kolibri/core/assets/src/views/CoreBase.vue

```diff
@@ -100,10 +97,10 @@
         required: false,
         default: '',
       },
-      // Prop that determines whether to show nav components
-      navBarNeeded: {
+      // Prop that determines whether to show nav components and provide margins
+      fullScreen: {
```

Comment:
Seems this introduces a new flavor `core-base`:
- `fullScreen`
- `immersivePage`
- "regular"

You think it's worth revisiting our approach to setting that "mode"? 

I remember some ideas floating around about transitioning to a vuex module approach? Seems to be the direction that https://github.com/learningequality/kolibri/pull/4495 was headed in.

Reply:
This was mostly just renaming `navBarNeeded` to `fullScreen` so that it could live up to its full potential. It allowed me to delete all this special-case stuff: https://github.com/learningequality/kolibri/pull/4588/files#diff-ab033c27dbeccc48ef01312c1fee6894L3

Reply:
> You think it's worth revisiting our approach to setting that "mode"?

I don't think so, no. Current implementation seems to work fine after this change